plugins/experimental/recursive_remove.cpp

```diff
@@ -0,0 +1,61 @@
+
+#include "experimental_plugin_framework.hpp"
+#include "parallel_remove_collection.hpp"
+
+#include "thread_pool.hpp"
+#include "connection_pool.hpp"
+
+#define IRODS_FILESYSTEM_ENABLE_SERVER_SIDE_API
+#include "filesystem.hpp"
+
+namespace irods::experimental::api {
+    namespace fs   = irods::experimental::filesystem;
+    namespace fsvr = irods::experimental::filesystem::server;
+
+    class recursive_remove : public base {
+
+        bool enable_asynchronous_operation() { return true; }
+
+        auto operation(rsComm_t* comm, const json req) -> json
+        {
+            try {
+                blackboard.set({{constants::status, states::running}});
+
+                fs::path path = get<std::string>("logical_path", req);
+
+                auto unregister = get<bool>("unregister", req, false);
+
+                auto no_trash = get<bool>("no_trash", req, false);
+
+                auto thread_count = get<int> ("thread_count", req, 4);
```

Comment:
why magic 4?

Reply:
this could come from the server config or some other place.